Add tests for vault + sdb

gtmanfred opened the pull request:
### What does this PR do?
Add tests for vault module and vault sdb

### What issues does this PR fix or reference?
Closes #46052

### Tests written?

Yes

### Commits signed with GPG?

Yes

gtmanfred committed:
add vault modules test

gtmanfred committed:
add sdb tests for vault

gtmanfred committed:
add sdb config option

gtmanfred committed:
don't include sdb:// in master/minion configs

cachedout commented on tests/integration/sdb/test_vault.py:
```diff
@@ -0,0 +1,63 @@
+        self.run_state('docker_image.absent', name='vault', force=True)
+
+    def test_sdb(self):
+        assert self.run_function('sdb.set', arg=['sdb://secret/test/secret/foo', 'bar']) is True
```
Perhaps we should make this path reference this test name in case we ever have to go back and debug something.

gtmanfred committed:
fix sdb test for vault

gtmanfred committed:
add running sdb tests

gtmanfred committed:
mark vault tests as flaky

The first couple commands to vault seem to be failing sometimes, but
after that this all works.

Commit:
Merge branch 'develop' into sdb

cachedout commented:
Nicely done.
